webclient/dokument/views.py

```diff
@@ -1606,36 +1638,50 @@ def archivovat(request, ident_cely):
         fedora_transaction = dokument.create_transaction(request.user, DOKUMENT_USPESNE_ARCHIVOVAN)
         dokument.active_transaction = fedora_transaction
         old_ident = dokument.ident_cely
-        # Nastav identifikator na permanentny
-        if ident_cely.startswith(IDENTIFIKATOR_DOCASNY_PREFIX):
-            rada = get_dokument_rada(dokument.typ_dokumentu, dokument.material_originalu)
-            try:
-                dokument.set_permanent_ident_cely(dokument.ident_cely[2], rada)
-            except MaximalIdentNumberError:
-                fedora_transaction.error_message = MAXIMUM_IDENT_DOSAZEN
-                fedora_transaction.rollback_transaction()
-                dokument.close_active_transaction_when_finished = True
-                return JsonResponse({"redirect": get_detail_json_view(ident_cely)}, status=403)
-            else:
-                dokument.save()
-                logger.debug("dokument.views.archivovat.permanent", extra={"ident_cely": dokument.ident_cely})
-        dokument.set_archivovany(request.user, old_ident)
-        dokument.doi_publish()
-        dokument.set_doi()
-        if dokument.rada == Heslar.objects.get(id=DOKUMENT_RADA_DATA_3D):
-            Mailer.send_ek01(document=dokument)
-        dokument.close_active_transaction_when_finished = True
-        dokument.save()
-        for item in dokument.casti.all():
-            item: DokumentCast
-            if (
-                item.archeologicky_zaznam
-                and item.archeologicky_zaznam.typ_zaznamu == ArcheologickyZaznam.TYP_ZAZNAMU_LOKALITA
-                and item.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY
-                and item.archeologicky_zaznam.lokalita.igsn
-            ):
-                item.archeologicky_zaznam.lokalita.igsn_update()
-        return JsonResponse({"redirect": get_detail_json_view(dokument.ident_cely)})
+        try:
+            for item in dokument.casti.all():
+                item: DokumentCast
+                if (
+                    item.archeologicky_zaznam
+                    and item.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY
+                    and item.archeologicky_zaznam.lokalita.igsn
+                ):
+                    item.archeologicky_zaznam.lokalita.igsn_update()
```

Comment:
Toto jsi nějak nově přesunul, což se opět myslím nejde, protože by se to snažilo udělat udate IGSN v době, kdy ještě dokument nemusí mít správné ID a navíc ještě nebude ve stavu D3. Musí to tedy myslím zůstat až před tím return.

Reply:
Přesunuto za volání metody `set_doi()`.

Reply:
Volání `save()` je v metodě `set_archivovany()`.